File: C++/1._Two_Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int> m;
        for(int i=0;i<nums.size();i++){
            int a = nums[i];
            int req= target - a;
            if(m.find(req)!=m.end()){
                return {m[req],i};
            }
            m[a]=i;
        }

        return nums;
    }
};

File: JavaScript/1._Two_Sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    for(let i=0;i<nums.length;i++){
        let req = target - nums[i];
        let index = nums.indexOf(req);
        if(index!==-1 && index!==i){
            return [i,index];
        }
    }
};